UPGRADE.md

```diff
@@ -1,10 +1,34 @@
 # Upgrade Guide
 
-_After `7.4.0` all `MINOR` and `MAJOR` version bumps will have upgrade notes 
-posted here._
+_`MAJOR` version bumps will have upgrade notes posted here._
+
+[2020-10-01] 7.x.x to 8.x.x
+-----------------------------
+
+### Overview
+
+Version `8.x.x` is the first version that officially drops support for Java 7. 
+
+- Removal of dependencies offering functionality included in Java 8 and beyond:
+    - Guava, for example:
+        - `com.google.common.base.Joiner` -> `String.join()`
+        - `com.google.common.collect.Lists` -> `Array.asList()`
+        - `com.google.common.base.Function` -> `java.util.function.Function`
```

Comment:
Need `Range` with example of how to migrate.

Reply:
I'm working on this one (`Range`) in this sprint with DI-850.